MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -50,16 +57,71 @@ class QuestionFactory: QuestionFactoryProtocol{
             text: "Рейтинг этого фильма больше чем 6?",
             correctAnswer: false)
     ]
+    */
     
-    weak var delegate: QuestionFactoryDelegate?
+//    func loadData() {
+//            moviesLoader.loadMovies { [weak self] result in
+//                guard let self = self else { return }
+//                switch result {
+//                case .success(let mostPopularMovies):
+//                    self.movies = mostPopularMovies.items // сохраняем фильм в нашу новую переменную
+//                    self.delegate?.didLoadDataFromServer() // сообщаем, что данные загрузились
+//                case .failure(let error):
+//                    self.delegate?.didFailToLoadData(with: error) // сообщаем об ошибке нашему MovieQuizViewController
+//                }
+//            }
+//        }
+    
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
Здесь не приведет к ошибке и утечке памяти, однако же рекомендую всегда использовать слабый захват в замыканиях и переослаблять ссылку в замыканиях вложенных
```suggestion
        moviesLoader.loadMovies { [weak self] result in
```